.github/workflows/build-on-pull-request.yml

```
@@ -8,11 +8,17 @@ jobs:
   Build:
     runs-on: ubuntu-latest
     steps:
-      - name: Checkout code
+      - name: Checkout code with submodules
         uses: actions/checkout@v3
```

actionlint (1.7.4)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)